ansible/roles/maintenance/tasks/main.yml

```diff
@@ -0,0 +1,30 @@
+---
+- name: Checkout Candlepin
+  git:
+    repo: "{{candlepin_git_repo}}"
+    dest: "{{candlepin_checkout}}"
+    force: yes
+    version: "{{candlepin_branch}}"
+    refspec: '+refs/pull/*:refs/heads/*'
+  when: "{{candlepin_git_pull}}"
+
+- name: Checkout Caracalla
+  git:
+    repo: "{{caracalla_git_repo}}"
+    dest: "{{caracalla_checkout}}"
+    force: yes
+    version: "{{caracalla_branch}}"
+  when: "{{candlepin_git_pull}}"
+
+- name: Update mysql vm post reset
+  shell: mysql -h {{mysql_vm_hostname}} -u candlepin candlepin < {{caracalla_checkout}}/ansible/resources/import.sql
+
+- name: Clear liquibase checksum
+  shell: "liquibase --driver=com.mysql.jdbc.Driver --classpath=server/src/main/resources:server/target/classes/:/usr/share/java/mysql-connector-java.jar --changeLogFile=db/changelog/changelog-update.xml --url=\"jdbc:mysql://{{mysql_vm_hostname}}:3306/candlepin\" --username=candlepin clearChecksums"
+  args:
+    chdir: "{{candlepin_checkout}}"
+
+- name: Update liquibase checksum
+  shell: "liquibase --driver=com.mysql.jdbc.Driver --classpath=server/src/main/resources:server/target/classes/:/usr/share/java/mysql-connector-java.jar --changeLogFile=db/changelog/changelog-update.xml --url=\"jdbc:mysql://{{mysql_vm_hostname}}:3306/candlepin\" --username=candlepin update -Dcommunity=true"
+  args:
+    chdir: "{{candlepin_checkout}}"
```

Comment:
So given how much of this file is duplicate code from the `candlepin-user` role, we should try to re-use, and not duplicate.

Organizationally, we have two overall use cases right now, running the tests, and making snapshots. Each are handled by separate playbooks. Great! Though, it would be more clear if we renamed `candlepin.yml` to `test.yml`.

Within those two playbooks there, is quite a bit of overlap... if you look at `ansible/roles/candlepin-user/tasks/main.yml`, you can decompose the tasks into different groups. For example, you might do:
 - checkout
 - configure
 - provision fresh database
 - deploy candlepin
 - run tests

If we do that, then we could reorganize a bit by decomposing `main.yml` into `tasks/checkout.yml`, `tasks/configure.yml`, etc., and then `main.yml` should look something like this:

```yaml
---
- include: checkout.yml

- include: configure.yml
  when: configure_candlepin

- include: provision_fresh_db.yml
  when: provision_fresh_db

- include: deploy.yml
  when: deploy_candlepin

- include: test.yml
  when: run_tests
```
Notice the use of `when` above; ideally we always have vars gating each include unless we're always going to run a set of tasks, such as the checkout tasks... then in `candlepin.yml`, you'd set the vars on the role as necessary (I'd argue we go ahead and default `configure_candlepin` to `true` in `defaults/main.yml`... and then override any defaults when you assign the role in the playbook, so for `candlepin.yml`:

```yaml
{ role: candlepin-user , provision_fresh_db: true, deploy_candlepin: true, run_tests: true }
```

and then for `maintenance.yml`:
```yaml
{ role: candlepin-user, configure_candlepin: false, provision_fresh_db: true }
```

To summarize:

I think the following should happen in this PR:
1. [ ] decompose `ansible/roles/candlepin-user/tasks/main.yml`.
2. [ ] add vars to control what happens when including the role (defaults should be idempotent).
3. [ ] remove now-redundant maintenance role.
3. [ ] change the role includes appropriately.

I think the following should happen, but can happen outside the scope of this PR.
1. [ ] `candlepin.yml` should be renamed `test.yml` or similar for clarity.
2. [ ] `candlepin-user` should probably be called something else, since it's also running jmeter tests...

As an aside, IMO it would be nice to eventually only have the idempotent tasks and data in roles, and have more of the tasks present in playbooks that we use to compose other playbooks as this feels more idiomatic, but let's not worry too much about that just yet.